src/ad-helper.js

```diff
@@ -82,13 +82,13 @@ export function isAdPositionAllowed(element, win) {
  */
 export function getAdContainer(element) {
   if (element[AD_CONTAINER_PROP] === undefined) {
-    let el = element;
-    do {
-      el = el.parentElement;
+    let el = element.parentElement;
+    while (el && el.tagName != 'BODY') {
```

Comment:
a safer looping 👍 

shall we remove `el.tagName != 'BODY'` as well to reduce confusion?

Reply:
I wasn't sure how important that might be. My hunch says that if we remove this the loop should at most run one more iteration (for the `<html>` element) but how does this work with shadow roots in all browsers? I feel like it's safer to leave this in unless someone more knowledgeable than me says otherwise :)